api/storage_service.cc

```diff
@@ -1000,7 +999,7 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
             return keyspaces;
         }
         const auto want_tablets = replication == "tablets";
-        return boost::copy_range<std::vector<sstring>>(keyspaces | boost::adaptors::filtered([&ctx, want_tablets] (const sstring& ks) {
+        return std::ranges::to<std::vector>(keyspaces | std::views::filter([&ctx, want_tablets] (const sstring& ks) {
```

Comment:
it is necessary. actually this is a two-fold question:

1. shall we preserve the existing behavior as much as possible. i think the answer is "yes". i don't intend to change the existing behavior. if it's changed by any chance, it's not on purpose. it's due to the inherent property of ranges library.
2. can we return a range in the function passed to `path_description::set(..)`. the answer is "no". `json_request_function` should return a `json::json_return_type`, which in turn should be constructed from any variable which can be formatted using `seastar::json::formatter::to_json()`. but this helper does not support ranges at this moment. see https://github.com/scylladb/seastar/blob/97e202afb14b58f0867a7fdd6a3deee8d648c822/include/seastar/json/formatter.hh#L198-L211 .

Reply:
@tchaikov sadly I cannot follow the details. I guess you're right because the compiler wouldn't compile something that can't work. I hope that someone with better familiarity with these ranges idioms can review this patch. I guess I missed the "functional programming" course that everyone else took at school ;-)

Reply:
thank you anyway. FWIW, i created https://github.com/scylladb/seastar/pull/2541 to enable `seastar::json::formatter::to_json()` to format a range.